### PR TITLE
BACK-926: new balancer v2 linear pools

### PR DESCRIPTION
**Pull request by aburkut**
Orignal PR from @johngrantuk https://github.com/paraswap/paraswap-dex-lib/pull/306

### Original PR description:

> We are releasing new Linear Pool types which will expand the Boosted Pools product to other protocols other than Aave.
> These pools perform exactly the same as the existing "AaveLinear" pools but will be named differently in the Subgraph.
> This PR adds the new types that will be part of the initial launch.
>  EulerLinear
>  BeefyLinear
> GearboxLinear
> MidasLinear
> ReaperLinear
> SiloLinear
> TetuLinear
> YearnLinear

### Summary:
- [x] add e2e tests for GearboxLinear pool which is the only available pool type from subgraph https://github.com/paraswap/paraswap-dex-lib/pull/333#issuecomment-1467979688
- [x] exclude pools with specific addresses from graphql query as it was suggested by @johngrantuk in comment https://github.com/paraswap/paraswap-dex-lib/pull/333#issuecomment-1467814063
- [x] deploy to Test env and run e2e tests against Test env API https://github.com/paraswap/paraswap-dex-lib/pull/333#issuecomment-1468225369

### CODE REVIEW

**Comment by aburkut** [comment]
GearboxLinear pool e2e test results:
paraswap-dex-lib – balancer-v2-e2e test ts 2023-03-14 15-06-09

**Comment by aburkut** [comment]
GearboxLinear pool e2e test results against Test env API:
paraswap-dex-lib –  env 2023-03-14 17-39-02

**Review by paraschiv-razvan-toptal — APPROVED** [review]

File: src/dex/balancer-v2/balancer-v2.ts [hunk 1]
```diff
@@ -62,12 +62,32 @@ const fetchAllPools = `query ($count: Int) {
     first: $count
     orderBy: totalLiquidity
     orderDirection: desc
-    where: {totalLiquidity_gt: ${MIN_USD_LIQUIDITY_TO_FETCH.toString()}, totalShares_not_in: ["0", "0.000000000001"], id_not_in: ["0xbd482ffb3e6e50dc1c437557c3bea2b68f3683ee0000000000000000000003c6"], swapEnabled: true, poolType_in: ["MetaStable", "Stable", "Weighted", "LiquidityBootstrapping", "Investment", "StablePhantom", "AaveLinear", "ERC4626Linear", "Linear", "ComposableStable"]}
+    where: {
```
Comment (Verisana) [inline src/dex/balancer-v2/balancer-v2.ts:65]: Cool! That is much easier to read :)

File: src/dex/balancer-v2/balancer-v2.ts [hunk 2]
```diff
@@ -62,12 +62,32 @@ const fetchAllPools = `query ($count: Int) {
...
+        "ERC4626Linear", "Linear", "ComposableStable", "BeefyLinear", "GearboxLinear", "MidasLinear",
+        "ReaperLinear", "SiloLinear", "TetuLinear", "YearnLinear"
+      ]
+    }
   ) {
     id
     address
     poolType
-    tokens {
+    tokens (orderBy: index) {
```
Comment (Verisana) [inline src/dex/balancer-v2/balancer-v2.ts:90]: @mwamedacen , look. There was another quick solution for the problem we had with indexes :smiley:

**Review by Verisana — APPROVED** [review]
Looks good to me

**Pull request merged**